### ISSUE TITLE
Wrong MIDI channels after export to MIDI file

### ISSUE THREAD

**Issue by AndrewPy** [ahlstromcj/seq66#109]
The channels are incorrect after exporting to a MIDI file. (This happens with Song Export, MIDI Only Export and SMF0 Export.)

For example, if you try it  with the attached file in seq66, all tracks except the drums will be exported on MIDI channel 1.

(In seq64 the export works correctly.)

[test1.zip](https://github.com/ahlstromcj/seq66/files/11324298/test1.zip)

**Comment by ahlstromcj** [comment]
I have verified this bug and will look at the problem soon. Thanks!

**Comment by ahlstromcj** [comment]
I believe I have fixed this issue (in the portfix branch).  I should test a
little more, but feel free to rebuild from that branch and let me know
of further issues.  The problem with the old exports was a brainfart,
while the issue with SMF 0 was with not applying the pattern's specified
channels to the events themselves.  I also had some "modified" issues that
cropped up, and I hope I did not break anything fixing them.
